File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Core/AlibabaCloud/SDK/ROS/CDK/Core/TokenComparison.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Core
{
    /// <summary>An enum-like class that represents the result of comparing two Tokens.</summary>
    /// <remarks>
    /// The return type of <c>Token.compareStrings</c>.
    /// </remarks>
    [JsiiClass(nativeType: typeof(AlibabaCloud.SDK.ROS.CDK.Core.TokenComparison), fullyQualifiedName: "@alicloud/ros-cdk-core.TokenComparison")]
    public class TokenComparison : DeputyBase
    {
        /// <summary>Used by jsii to construct an instance of this class from a Javascript-owned object reference</summary>
        /// <param name="reference">The Javascript-owned object reference</param>
        [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
        protected TokenComparison(ByRefValue reference): base(reference)
        {
        }

        /// <summary>Used by jsii to construct an instance of this class from DeputyProps</summary>
        /// <param name="props">The deputy props</param>
        [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
        protected TokenComparison(DeputyProps props): base(props)
        {
        }

        /// <summary>This means both components are Tokens.</summary>
        [JsiiProperty(name: "BOTH_UNRESOLVED", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.TokenComparison\"}")]
        public static AlibabaCloud.SDK.ROS.CDK.Core.TokenComparison BOTH_UNRESOLVED
        {
            get;
        }
        = GetStaticProperty<AlibabaCloud.SDK.ROS.CDK.Core.TokenComparison>(typeof(AlibabaCloud.SDK.ROS.CDK.Core.TokenComparison))!;

        /// <summary>This means we're certain the two components are NOT Tokens, and different.</summary>
        [JsiiProperty(name: "DIFFERENT", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.TokenComparison\"}")]
        public static AlibabaCloud.SDK.ROS.CDK.Core.TokenComparison DIFFERENT
        {
            get;
        }
        = GetStaticProperty<AlibabaCloud.SDK.ROS.CDK.Core.TokenComparison>(typeof(AlibabaCloud.SDK.ROS.CDK.Core.TokenComparison))!;

        /// <summary>This means exactly one of the components is a Token.</summary>
        [JsiiProperty(name: "ONE_UNRESOLVED", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.TokenComparison\"}")]
        public static AlibabaCloud.SDK.ROS.CDK.Core.TokenComparison ONE_UNRESOLVED
        {
            get;
        }
        = GetStaticProperty<AlibabaCloud.SDK.ROS.CDK.Core.TokenComparison>(typeof(AlibabaCloud.SDK.ROS.CDK.Core.TokenComparison))!;

        /// <summary>This means we're certain the two components are NOT Tokens, and identical.</summary>
        [JsiiProperty(name: "SAME", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.TokenComparison\"}")]
        public static AlibabaCloud.SDK.ROS.CDK.Core.TokenComparison SAME
        {
            get;
        }
        = GetStaticProperty<AlibabaCloud.SDK.ROS.CDK.Core.TokenComparison>(typeof(AlibabaCloud.SDK.ROS.CDK.Core.TokenComparison))!;
    }
}

File: multiple-languages/java/ros-cdk-core/src/main/java/com/aliyun/ros/cdk/core/TokenComparison.java
package com.aliyun.ros.cdk.core;

/**
 * An enum-like class that represents the result of comparing two Tokens.
 * <p>
 * The return type of <code>Token.compareStrings</code>.
 */
@javax.annotation.Generated(value = "jsii-pacmak/1.85.0 (build 08ee592)", date = "2025-08-12T08:28:43.030Z")
@software.amazon.jsii.Jsii(module = com.aliyun.ros.cdk.core.$Module.class, fqn = "@alicloud/ros-cdk-core.TokenComparison")
public class TokenComparison extends software.amazon.jsii.JsiiObject {

    protected TokenComparison(final software.amazon.jsii.JsiiObjectRef objRef) {
        super(objRef);
    }

    protected TokenComparison(final software.amazon.jsii.JsiiObject.InitializationMode initializationMode) {
        super(initializationMode);
    }

    static {
        BOTH_UNRESOLVED = software.amazon.jsii.JsiiObject.jsiiStaticGet(com.aliyun.ros.cdk.core.TokenComparison.class, "BOTH_UNRESOLVED", software.amazon.jsii.NativeType.forClass(com.aliyun.ros.cdk.core.TokenComparison.class));
        DIFFERENT = software.amazon.jsii.JsiiObject.jsiiStaticGet(com.aliyun.ros.cdk.core.TokenComparison.class, "DIFFERENT", software.amazon.jsii.NativeType.forClass(com.aliyun.ros.cdk.core.TokenComparison.class));
        ONE_UNRESOLVED = software.amazon.jsii.JsiiObject.jsiiStaticGet(com.aliyun.ros.cdk.core.TokenComparison.class, "ONE_UNRESOLVED", software.amazon.jsii.NativeType.forClass(com.aliyun.ros.cdk.core.TokenComparison.class));
        SAME = software.amazon.jsii.JsiiObject.jsiiStaticGet(com.aliyun.ros.cdk.core.TokenComparison.class, "SAME", software.amazon.jsii.NativeType.forClass(com.aliyun.ros.cdk.core.TokenComparison.class));
    }

    /**
     * This means both components are Tokens.
     */
    public final static com.aliyun.ros.cdk.core.TokenComparison BOTH_UNRESOLVED;

    /**
     * This means we're certain the two components are NOT Tokens, and different.
     */
    public final static com.aliyun.ros.cdk.core.TokenComparison DIFFERENT;

    /**
     * This means exactly one of the components is a Token.
     */
    public final static com.aliyun.ros.cdk.core.TokenComparison ONE_UNRESOLVED;

    /**
     * This means we're certain the two components are NOT Tokens, and identical.
     */
    public final static com.aliyun.ros.cdk.core.TokenComparison SAME;
}
